### PR TITLE
Feat: Setting up Testing tools

### PR DESCRIPTION
**Pull request by jonathanperazawl**
## Description

This PR integrates basic configuration for linting with typescript base rules and unit testing with Jest + React Testing Library and Functional testing with Cypress.io

## How to test

* Run it locally
* To test commitlint you can run `echo "this is a test message" | ./node_modules/.bin/commitlint `,  it would throw an error and the message shouldn't be saved on staged files.
* To test lint: `yarn lint`
* To run unit tests: `yarn test`, it must run a single unit test for the homepage
* To run e2e: `yarn e2e`, it will open a dialog window in which you need to click on `run all`
image

### CODE REVIEW

**Comment by ivanovishado** [comment]
Before reviewing it, I'm thinking that you can add some of the instructions you wrote on this PR's description to the README.

Edit: and looking at the `commitlint`, maybe also add a commit format to the `contributing.md` file, but that can also be on the README, I don't know where's the best place to put it.

**Comment by jonathanperazawl** [comment]
Thanks @ivanovishado I was thinking about that, but I will address on a different PR in which we can set all the instructions for contribution and best practices, what do you think?

**Review by ivanovishado — APPROVED** [review]
Ok, the configuration looks good. Although it's throwing some errors when building, will this be addressed on another PR as well?

**Comment by jonathanperazawl** [comment]
@ivanovishado  I didn't test the building, let me check

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"]
```
Comment (jackbravo) [inline .babelrc:2]: Just curious. Why is babel or this file needed?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  rules: {
+    'react/prop-types': 'off', // We turn off prop-types rule, as we will use TypeScript's types instead.
+    "react/react-in-jsx-scope": "off",
+    // allow jsx syntax in js files (for next.js project)
+   "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".tx", ".tsx"] }],
+  },
+  "env": {
+    "jest": true
+  }
+};
```
Comment (jackbravo) [inline .eslintrc.js:24]: I like eslint. But I like more prettier. But I guess for this project it is better to have suggestions that you have to manually fix, to reinforce learning, than to have a tool automatically applying changes for you.

File: package.json [hunk 3]
```diff
@@ -3,24 +3,51 @@
...
     "build": "next build",
     "start": "next start",
-    "storybook": "start-storybook"
+    "storybook": "start-storybook",
+    "lint": "eslint ./components** ./pages/** -c .eslintrc.js --ext ts,tsx",
+    "test": "jest",
+    "e2e": "cypress open"
   },
   "dependencies": {
+    "cypress": "^5.0.0",
```
Comment (jackbravo) [inline package.json:15]: Shouldn't this be a dev dependency?

**Review by jackbravo — APPROVED** [review]

**Comment by jonathanperazawl** [comment]
Sorry for the late response, I'll address the comments on a different PR just for demo's purpose

**Pull request merged**